### DISCUSSION TITLE
DLNA / uPNP network source (play DLNA streams in rAudio-1)

### DISCUSSION THREAD

**Discussion by subatomicglue** [rern/rAudio#602] [Feature Requests]
Hi, looks like rAudio-1 can be a DLNA/uPNP server.   

But I'd love it to be a client, to play stuff off my DLNA/uPNP server (I have a readynas with a huge music collection).

Background:   In my house, it's how I've been playing audio before finding "rAudio-1", using the mconnectLite or AirPlayer dlna clients for iPhone...  So pointing rAudio-1 at my DLNA server would be nice for me... :)

I realize for now I can use NFS or CIFS...

Just curious if DLNA client was on the roadmap?

**Comment by findnix** [comment]
Tried this?

![image](https://user-images.githubusercontent.com/77154016/145275543-47b52b5a-bf46-4b87-b716-83d47fc92f66.png)

**Reply by subatomicglue** [reply]
Right, that seems to make rAudio1 be a DLNA server... (renderer means server (play to other DLNA clients), not client/player, right??).   The word Renderer is ambiguous to me here, hard to grok :)

Unless I'm mis understanding - totally possible.   Understanding this UI at times is not 100% straightforward, or easy to confuse myself here :)

So I guess the problem is that:
When I enabled that UPnP option I did NOT see any DLNA servers on my network show up in the library....

Does it work for you?
Are you able to have your rAudio1 device play from any of your DLNA servers?  If so, then what do I need to do after flicking this option to ON, to find my DLNA server and play from it???

**Comment by hjfisch** [comment]
You will not find your Music-Server in raudio , but you can use your UPNP Musicplayer App on your NAS , PC or Cellphone to stream directly to raudio . Otherwise you can connect your NAS as a Networkdrive.

**Reply by subatomicglue** [reply]
So, I was aware of that in the OP above.    I was asking for the feature for rAudio1 to add a client/player to play from Upnp/DLNA servers.   

My NAS only has a DLNA server, no client or app there that I can control.   My cellphone of course has a DLNA client/player, which uses my NAS DLNA server as a source of music files, but I have 0 reason to use my cellphone to push DLNA/Upnp  over to the rAudio, why wouldn't I use bluetooth or maybe airplay for that? - both of which is more intuitive on an iPhone...

Having rAudio navigate file lists coming from Upnp/DLNA servers would be useful, and fit with the functionality it already has to pull from NFS sources.     ........     Also important, probably, would be NOT having to index those sources (too slow to wait for the index to update for transient DLNA sources there...   I mean.... for NFS too, it's just horribly slow.  Maybe it's a separate feature request though :) ).

**Comment by elliotn** [comment]
If you use Android, Bubble UPNP is awesome (https://play.google.com/store/apps/details?id=com.bubblesoft.android.bubbleupnp). I have a music library on my NAS, which I access via Bubble UPNP*. From there I point to RuneAudio as a render. 

I've used this for quite some time on older official RuneAudio installs. I just updated to rAudio yesterday, which mostly works. I've noticed that I have had to stop/start or resend music (files or radio streams). I have yet to figure out if rAudio or Bubble UPNP is the problem. 

* I actually use minimserver as the DLNA server to index the files on the NAS and serve up the files. It does a much better job at indexing than my NAS.

**Reply by elliotn** [reply]
Also, I just discovered today that I can use Antenna Pod to Share a podcast episode with Bubble UPNP, which then sends it to rAudio!!!! Not sure I'll ever bother with bluetooth again.

**Comment by ceejayemm** [comment]
I agree with @subatomicglue, having rAudio-1 being able to act as a uPNP/DNLA Player (able to attach to, and play music from, existing uPNP/DNLA servers) without the need for an additional Control Point would be great. I proposed this to @rern some long time ago when we are using the old RE addons to RuneAudio, he didn't seem too keen at that point as in some respects the existing network share/MPD library/ (now) rAudio-1 Player do much the same thing - and its all in one installation (great plus point!). It can however be time consuming though when the Library on several RPi need to be updated eg following a reflash of the SD card with a new base version of rAudio-1.

I currently have my music collection on a OpenMediaVault v5.x NAS connected to each rAudio-1 installation via an NFS share. I have a MinimServer installation available as well which also works well with rAudio-1 if you have the other bits (control point / player) needed to make the connection. If rAudio-1 could include a control point to attach to an existing server(s) (perhaps as a Source ?) and send its output to the rAudio-1 player then that would make uPNP/DNLA a viable option for me too. 

It seems the rAudio-1 uPNP software (upmpdcli ) is already compatible with the existing MPD Library so (hopefully) it might mean the step is not be a very big one to make ? The benefit to me would be that MinimServer (in my case) would only need to be updated once when my store of music/radios changes and the changes would then be immediately served out to all my rAudio-1 installations making updates there much quicker

Chris

**Reply by subatomicglue** [reply]
Agreed, it's just odd to me that there would be CIFS or NFS on rAudio1, but no similar file reader for DLNA.

It's also odd to me to need to involve my smart phone at all.   Since... that's the whole point of this standalone device is as a kiosk.   So needing to use a DLNA player/client on my phone kind of defeats the whole purpose, I may as well not have an rAudio1 device then.

I have the NetGear ReadyNAS 626x  which has a DLNA server built in as well as Plex serving DLNA as well...  just as it has NFS and CIFS...   connecting to CIFS with rAudio was unstable though... idk what was happening, but it made browsing files sketchy and brittle, or not work. (100GB of mp3 and m4a in nested folders).  removing the CIFS connection allowed rAudio1 to browse music on my USB drive again.  I have a suspicion that the rAudio1 indexing function is a cancer that causes some of this nightmare though, since it processes in the background you can't ever know when it is done - or if it is done, or what's going on.  so My hope with DLNA is that there would be NO indexing, which would be a more stable file browsing experience :)

**Reply by ceejayemm** [reply]
All my music is on my NAS and I access it via NFS with no updating / indexing issues (other than time) at all - no USBs for me. Once I changed to using the IP Number of the NAS, rather than its network name, the network connections have been rock solid on all my Pi's (RPi Zero W, RPI 3A and RPI3 B+).

DNLA would be a 'nice to have' for me to centralize the library across all my machines and have the extra indexing options that something like MinimServer would give, but I can live with the current setup. An implementation of combined Control Point / Player in rAudio-1 would be interesting but I can also see the one-stop benefits of a single implementation that Audio-12 currently is. The remote app I use on my Android phone when I can't get to a PC/Browser also works well but I agree I don't want to have to use this continuously to send music to rAudio-1 to play.

Happy to test any future implementations but I think rAudio-1 is pretty good as it is and it is my goto application for listening to music or radio stations - thanks @rern

Chris

**Reply by subatomicglue** [reply]
```
thanks @rern
```
Agreed, much gratitude to @rern

**Reply by elliotn** [reply]
Got it re: kiosk. I have a couple of rune audio devices that I treat as hidden music playing appliances.
